src/components/Modals/FiatRamps/views/Manager.tsx

```diff
@@ -55,132 +55,133 @@ const moduleLogger = logger.child({
-  const [ensName, setEnsName] = useState<string>('')
-
-  const chainAdapterManager = getChainAdapterManager()
-  const ethereumChainAdapter = chainAdapterManager.get(KnownChainIds.EthereumMainnet)
-  const avalancheChainAdapter = chainAdapterManager.get(KnownChainIds.AvalancheMainnet)
-  const bitcoinChainAdapter = chainAdapterManager.get(KnownChainIds.BitcoinMainnet)
-  const bitcoinCashChainAdapter = chainAdapterManager.get(KnownChainIds.BitcoinCashMainnet)
-  const dogecoinChainAdapter = chainAdapterManager.get(KnownChainIds.DogecoinMainnet)
-  const litecoinChainAdapter = chainAdapterManager.get(KnownChainIds.LitecoinMainnet)
-  const cosmosChainAdapter = chainAdapterManager.get(KnownChainIds.CosmosMainnet)
-
-  const [chainId, setChainId] = useState<ChainId>(ethChainId)
+  const [accountId, setAccountId] = useState<Nullable<AccountId>>(null)
+  const [addressByAccountId, setAddressByAccountId] = useState<AddressesByAccountId>({})
 
   const {
     state: { wallet },
   } = useWallet()
-  const [accountId, setAccountId] = useState<Nullable<AccountId>>(null)
 
+  /**
+   * preload all addresses, and reverse resolved vanity addresses for all account ids
+   */
   useEffect(() => {
+    if (!wallet) return
     ;(async () => {
-      if (!wallet) return
-      moduleLogger.trace({ fn: 'getAddress' }, 'Getting Addresses...')
-      const payload = { wallet }
-      try {
-        if (supportsETH(wallet) && ethereumChainAdapter) {
-          setEthAddress(await ethereumChainAdapter.getAddress(payload))
-        }
-        if (supportsEthSwitchChain(wallet) && avalancheChainAdapter) {
-          setAvalancheAddress(await avalancheChainAdapter.getAddress(payload))
-        }
-        if (supportsBTC(wallet) && bitcoinChainAdapter) {
-          setBtcAddress(await bitcoinChainAdapter.getAddress(payload))
-        }
-        if (supportsBTC(wallet) && bitcoinCashChainAdapter) {
-          setBchAddress(await bitcoinCashChainAdapter.getAddress(payload))
-        }
-        if (supportsBTC(wallet) && dogecoinChainAdapter) {
-          setDogeAddress(await dogecoinChainAdapter.getAddress(payload))
-        }
-        if (supportsBTC(wallet) && litecoinChainAdapter) {
-          setLtcAddress(await litecoinChainAdapter.getAddress(payload))
-        }
-        if (supportsCosmos(wallet) && cosmosChainAdapter) {
-          setCosmosAddress(await cosmosChainAdapter.getAddress(payload))
+      const plainAddressResults = await Promise.allSettled(
+        portfolioAccountIds.map(accountId => {
+          const accountMetadata = portfolioAccountMetadata[accountId]
+          const { accountType, bip44Params } = accountMetadata
+          moduleLogger.trace({ fn: 'getAddress' }, 'Getting Addresses...')
+          const payload = { accountType, bip44Params, wallet }
+          const { chainId } = fromAccountId(accountId)
+          return getChainAdapterManager().get(chainId)!.getAddress(payload)
```

Comment:
In this case with our current business logic yes - an account id is only obtained from a chain adapter. Unless a user turns a flag off at runtime it will still be valid, but I will change this for safety.